pkg/cmd/server/origin/master.go

```diff
@@ -285,45 +276,10 @@ func (c *MasterConfig) buildHandlerChain() (func(apiHandler http.Handler, kc *ap
 
-			// these are all equivalent to the kube handler chain
-			///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
-			handler = serverhandlers.AuthorizationFilter(handler, c.Authorizer, c.AuthorizationAttributeBuilder, genericConfig.RequestContextMapper)
-			handler = serverhandlers.ImpersonationFilter(handler, c.Authorizer, cache.NewGroupCache(c.UserInformers.User().InternalVersion().Groups()), genericConfig.RequestContextMapper)
-			// audit handler must comes before the impersonationFilter to read the original user
-			if c.Options.AuditConfig.Enabled {
-				var writer io.Writer
-				if len(c.Options.AuditConfig.AuditFilePath) > 0 {
-					writer = &lumberjack.Logger{
-						Filename:   c.Options.AuditConfig.AuditFilePath,
-						MaxAge:     c.Options.AuditConfig.MaximumFileRetentionDays,
-						MaxBackups: c.Options.AuditConfig.MaximumRetainedFiles,
-						MaxSize:    c.Options.AuditConfig.MaximumFileSizeMegabytes,
-					}
-				} else {
-					// backwards compatible writer to regular log
-					writer = cmdutil.NewGLogWriterV(0)
-				}
-				c.AuditBackend = auditlog.NewBackend(writer)
-				auditPolicyChecker := auditpolicy.NewChecker(&auditinternal.Policy{
-					// This is for backwards compatibility maintaining the old visibility, ie. just
-					// raw overview of the requests comming in.
-					Rules: []auditinternal.PolicyRule{{Level: auditinternal.LevelMetadata}},
-				})
-				handler = apifilters.WithAudit(handler, genericConfig.RequestContextMapper, c.AuditBackend, auditPolicyChecker, genericConfig.LongRunningFunc)
-			}
-			handler = apifilters.WithAuthentication(handler, c.RequestContextMapper, c.Authenticator, apifilters.Unauthorized(false))
-			handler = apiserverfilters.WithCORS(handler, c.Options.CORSAllowedOrigins, nil, nil, nil, "true")
-			handler = apiserverfilters.WithTimeoutForNonLongRunningRequests(handler, genericConfig.RequestContextMapper, genericConfig.LongRunningFunc)
-			// TODO: MaxRequestsInFlight should be subdivided by intent, type of behavior, and speed of
-			// execution - updates vs reads, long reads vs short reads, fat reads vs skinny reads.
-			// NOTE: read vs. write is implemented in Kube 1.6+
-			handler = apiserverfilters.WithMaxInFlightLimit(handler, genericConfig.MaxRequestsInFlight, genericConfig.MaxMutatingRequestsInFlight, genericConfig.RequestContextMapper, genericConfig.LongRunningFunc)
-			handler = apifilters.WithRequestInfo(handler, apiserver.NewRequestInfoResolver(genericConfig), genericConfig.RequestContextMapper)
-			handler = apirequest.WithRequestContext(handler, genericConfig.RequestContextMapper)
-			handler = apiserverfilters.WithPanicRecovery(handler)
-			///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+			handler = apiserver.DefaultBuildHandlerChain(handler, genericConfig)
```

Comment:
@sttts as a past rebaser, this should make you very happy.

Reply:
❤️